### DISCUSSION TITLE
How to conditionally update a property?

### DISCUSSION THREAD

**Discussion by devdiegomatos** [livewire/livewire#5531] [1. Help] [answered]
Hi,

I have a Livewire Component called Question that can only take Answers once, after the answer is taken I set a boolean property _hasAnsweredQuestion_ to true and try to check on subsequent updatingAnswerId lifecycle request method if this property is checked.

```
public function updatingAnswerId(int $incomingValue): void
{
    $oldValue = $this->answerId;

    if ($this->hasAnsweredQuestion) {
        $this->answerId = $oldValue;
    } else {
        $this->answerId = $incomingValue;
    }
}
```

However the answerId property is alway set to the IncomingValue. I have already tried returning the desired value and other lifecycle methods such as: updated, hydrate and dehydrate. None of them works as expected. 

How to handle this situation?

**Comment by joshhanley** [comment, marked as answer]
@diegomatos321 you need to receive the `$incomingValue` by reference `&` and update it directly, like this:

```php
public function updatingAnswerId(int &$incomingValue): void
{
    if ($this->hasAnsweredQuestion) {
        $incomingValue = $this->answerId;
    }
}
```

Then Livewire will still process the value as per normal but using the old value.

Hope this helps!

**Reply by devdiegomatos** [reply]
Thank you Josh, it works! 

Just a note for the future, your explanation is correct but the given example has a mistaken, you forgot to add the '&' on the function's parameter, if someone struggle with that just take a look at the php documentation:

[https://www.php.net/manual/en/language.references.pass.php](https://www.php.net/manual/en/language.references.pass.php)

**Reply by joshhanley** [reply]
@diegomatos321 no worries! 🙂

Whoops! Yeah I forgot to add that in and it was the critical piece 😆 I have updated the code example in case anyone else comes across it.
